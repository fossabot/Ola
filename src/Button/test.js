import React from 'react'
import { Button } from './'
import renderer from 'react-test-renderer'

it('No variant button', () => {
  const tree = renderer
    .create(<Button>Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Primary button', () => {
  const tree = renderer
    .create(<Button variant="primary">Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Secondary button', () => {
  const tree = renderer
    .create(<Button variant="secondary">Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('PRO button', () => {
  const tree = renderer
    .create(<Button variant="pro">Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Primary destructive button', () => {
  const tree = renderer
    .create(<Button variant="destructive-primary">Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Destructive button', () => {
  const tree = renderer
    .create(<Button variant="destructive">Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Busy button', () => {
  const tree = renderer
    .create(<Button variant="primary" busy>Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Primary button with extra class', () => {
  const tree = renderer
    .create(<Button variant="primary" extraClass="is-extra">Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Disabled primary button', () => {
  const tree = renderer
    .create(<Button variant="primary" disabled>Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Disabled link button', () => {
  const tree = renderer
    .create(<Button variant="link" disabled>Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})