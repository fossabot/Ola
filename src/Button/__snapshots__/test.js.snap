// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Busy button 1`] = `
<button
  className="ola_button is-busy"
  disabled={true}
>
  <svg
    className="ola_spinner"
    height={28}
    viewBox="0 0 28 28"
    width={28}
  >
    <circle
      cx={14}
      cy={14}
      r={12}
    />
  </svg>
  <span
    className="ola_button-text"
  />
</button>
`;

exports[`Destructive button 1`] = `
<button
  className="ola_button is-destructive"
  disabled={false}
>
  <span
    className="ola_button-text"
  >
    Click here
  </span>
</button>
`;

exports[`Disabled primary button 1`] = `
<button
  className="ola_button is-primary"
  disabled={true}
>
  <span
    className="ola_button-text"
  >
    Click here
  </span>
</button>
`;

exports[`No variant button 1`] = `
<button
  className="ola_button"
  disabled={false}
>
  <span
    className="ola_button-text"
  >
    Click here
  </span>
</button>
`;

exports[`PRO button 1`] = `
<button
  className="ola_button is-pro"
  disabled={false}
>
  <span
    className="ola_button-text"
  >
    Click here
  </span>
</button>
`;

exports[`Primary button 1`] = `
<button
  className="ola_button is-primary"
  disabled={false}
>
  <span
    className="ola_button-text"
  >
    Click here
  </span>
</button>
`;

exports[`Primary button with extra class 1`] = `
<button
  className="ola_button is-primary is-extra"
  disabled={false}
>
  <span
    className="ola_button-text"
  >
    Click here
  </span>
</button>
`;

exports[`Primary destructive button 1`] = `
<button
  className="ola_button is-destructive-primary"
  disabled={false}
>
  <span
    className="ola_button-text"
  >
    Click here
  </span>
</button>
`;

exports[`Secondary button 1`] = `
<button
  className="ola_button is-secondary"
  disabled={false}
>
  <span
    className="ola_button-text"
  >
    Click here
  </span>
</button>
`;
